File: app/src/main/java/fr/xgouchet/chronorg/ui/viewholders/JumpViewHolder.java
package fr.xgouchet.chronorg.ui.viewholders;

import android.support.annotation.NonNull;
import android.view.View;
import android.widget.TextView;

import org.joda.time.format.DateTimeFormat;
import org.joda.time.format.DateTimeFormatter;

import butterknife.BindView;
import butterknife.OnClick;
import fr.xgouchet.chronorg.R;
import fr.xgouchet.chronorg.data.models.Jump;

import static butterknife.ButterKnife.bind;

/**
 * @author Xavier Gouchet
 */
public class JumpViewHolder extends BaseViewHolder<Jump> {

    final DateTimeFormatter dtf = DateTimeFormat.forStyle("MM").withZoneUTC();

    @BindView(R.id.name) TextView name;
    @BindView(R.id.from_instant) TextView from_instant;
    @BindView(R.id.to_instant) TextView to_instant;

    public JumpViewHolder(@NonNull Listener<Jump> listener, View itemView) {
        super(listener, itemView);
        bind(this, itemView);
    }

    @Override public void onBindItem(@NonNull Jump jump) {
        name.setText(jump.getName());

        from_instant.setText(dtf.print(jump.getFrom()));
        to_instant.setText(dtf.print(jump.getTo()));
    }

    @OnClick(R.id.jump)
    public void onSelectProject() {
        fireSelected();
    }
}

File: khronorg/src/main/java/fr/xgouchet/khronorg/ui/viewholders/JumpViewHolder.kt
package fr.xgouchet.khronorg.ui.viewholders

import android.view.View
import android.widget.TextView
import fr.xgouchet.khronorg.R
import fr.xgouchet.khronorg.data.formatters.Formatter
import fr.xgouchet.khronorg.data.models.Jump
import fr.xgouchet.khronorg.ui.Cutelry.knife
import io.reactivex.functions.Consumer
import org.joda.time.ReadableInstant

/**
 * @author Xavier F. Gouchet
 */
class JumpViewHolder(val formatter: Formatter<ReadableInstant>, listener: Consumer<Jump>?, itemView: View)
    : BaseViewHolder<Jump>(listener, itemView) {


    internal val from: TextView by knife(R.id.from_instant, itemView)
    internal val to: TextView by knife(R.id.to_instant, itemView)

    init {
        if (listener != null) {
            itemView.setOnClickListener({ _ -> listener.accept(item) })
        }
    }

    override fun onBindItem(item: Jump) {
        from.text = formatter.format(item.from)
        to.text = formatter.format(item.destination)
    }
}